packages/loot-core/src/shared/util.ts

```diff
@@ -362,3 +362,29 @@ export function sortByKey<T>(arr: T[], key: keyof T): T[] {
     return 0;
   });
 }
+
+export function looselyParseAdditionalInformation(additionalInformation) {
+  let additionalInformationJson;
+  try {
+    additionalInformationJson = JSON.parse(additionalInformation);
+  } catch (e) {
+    try {
+      // It was not valid JSON, attempt to make the payload valid JSON using regex.
+      let result = {};
+      let matches = additionalInformation.matchAll(
+        /(, )?([^:]+): ((\[.*?\])|([^,]*))/g,
+      );
+      for (let match of matches) {
+        let key = match[2].trim();
+        let value = (match[4] || match[5]).trim();
+        // Remove square brackets and single quotes and commas
+        value = value.replace(/[[\]',]/g, '');
+        result[key] = value;
+      }
+      additionalInformationJson = result;
+    } catch (e) {
+      additionalInformationJson = {};
+    }
+  }
+  return additionalInformationJson;
+}
```

Comment:
A couple of things:
1. I've never heard of a single case in which this field would contain serialized JSON, so trying to parse it as JSON seems pointless to me.
2. I'm not really a fan of using regex to try to parse something like this, especially without a specification of the actual format used that we can validate it against.  Your current regexes would break if e.g. one of the fields can contain `]`, which without a specification we don't know if it's valid.